cmd/tailscale/cli/exitnode.go

```diff
@@ -271,9 +272,14 @@ func filterFormatAndSortExitNodes(peers []*ipnstate.PeerStatus, filterBy string)
 			countryAnyPeer = append(countryAnyPeer, city.Peers...)
 			var reducedCityPeers []*ipnstate.PeerStatus
 			for i, peer := range city.Peers {
+				if filterBy != "" {
+					// If the peers are being filtered, we return all peers to the user.
+					reducedCityPeers = append(reducedCityPeers, city.Peers...)
+					break
```

Comment:
I feel like this kind of belongs outside the loop but then it's awkward in a different way. I suspect there's a better way to do this by adding just the first element or all elements in one pass, then at the end add in whatever the current exit node is.

Anyway this is fine for now, I'm not clear that another approach is going to save any complexity in the end.